### ISSUE TITLE
f2py build failure under numpy 1.16.0, 1.16.1 and 1.16.2

### ISSUE THREAD

**Issue by jtniehof** [spacepy/spacepy#90]
#88 had a failed build in Travis under fairly particular conditions. Not sure why this wasn't flagged as not-okay in the PR. There's an f2py failure: [travis log](https://travis-ci.com/spacepy/spacepy/builds/98668144)

Python 3.5, numpy 1.16.0 if I'm reading the logs properly.

**Comment by jtniehof** [comment]
Actually this is probably unrelated to #88 since it looks like a build well after the commit was merged. It failed starting 10 days ago. Looks like numpy 1.15.4 works and 1.16.0 fails. Funny how it's okay on Python 3.6 but not 3.5; I can't tell what version of numpy is installed in 3.6 (the pip output is a bit different).

**Comment by drsteve** [comment]
We should probably update the Travis scripts to make sure that the version info is printed out in the logs. Maybe a pip freeze?

Anyway, there's no obvious culprit in the numpy 1.16.0 [release notes](https://github.com/numpy/numpy/blob/master/doc/release/1.16.0-notes.rst). Although I note that they changed f2py on Windows from a Python script to an executable. Maybe for now we update the travis.yml to not test on 1.16 and set ourselves a deadline for either working out the issue or adding 1.16 (with Python 3.5) to our not supported list until we resolve the issue.

**Comment by balarsen** [comment]
I can go with this but I have reservations with limiting ourselves away from the newest numpy. I have been doing a lot of pymc3 lately and new versions there need both numpy 1.16 and new theano. This then makes spacepy with fortran incompatible with the setup. 

Not that I have time to do it, but maybe this is part of the kick to get the "spacepy-kits" going and move the fortran out there. Or make the move to LGM.

**Comment by drsteve** [comment]
I'm not saying write off new versions of numpy. There's potentially a
workaround on our end, and likely a fix in the works on the numpy end. But
having our tests failing due to a dependency regression shouldn't reflect
on our library. We should do what we can from our end, but ...

Breaking out parts into kits is definitely on the plan, but after the next
release. That would definitely help, but it's going to have to wait.

My guess is that there'll be a 1.16.2 fairly soon that fixes this. After
all, there was an early release of 1.15 that broke stuff too. Meanwhile, I
can't see a good reason for us to test against versions that have a known
issue with f2py. That just makes it look like SpacePy is failing its own
tests when we're not.
 And if we find a workaround then we can say we support these versions
again.

**Comment by drsteve** [comment]
Well, there's the bug & PR over at numpy that I referenced in the issue
tracker. And it's only 1.16(.0 and .1) that are failing for us, both with a
failure on calling f2py. Check out the Travis logs.

Python 3.6 was passing because the test was pulling greater than our
minimum version and excluding one specific version, but 3.6 was the only
install not grabbing numpy 1.16 (for some weird pip related reason).

Hence my suggestion for this PR and keeping the issue tracker open but
renaming to something like "f2py failures under numpy 1.16" so we can see
whether it's a string/byte issue that we can work around, and we can close
it whenever numpy fixes the regression (or we have a functional workaround).

**Comment by jtniehof** [comment]
Ah, if 3.6 wasn't running numpy 1.16, makes a lot more sense. I agree with blacklisting numpy versions that we know from elsewhere are likely to misbehave on f2py and continue to see if we can verify that's really our root cause here.

**Comment by drsteve** [comment]
Okay, given that I think we're all on the same page now, can I ask someone to go take a look at the PR (#91)?

**Comment by balarsen** [comment]
I would rather JOn do it as he has a better handle on the CI etc

**Comment by jtniehof** [comment]
I have no handle on the CI! But I'll take a look at the PR.

**Comment by drsteve** [comment]
Tests are failing again on the versions that are picking up numpy 1.16.2

I took a closer look at the numpy bug and PR referenced above and it looks like the resolution to that issue should have been in 1.16.1 already. So we really need to work out what broke in f2py and how to patch it on our end.

**Comment by drsteve** [comment]
Here's the relevant part of the traceback (from the Travis Python 3.7 build with numpy 1.16.2)

Am I right in thinking that the behaviour of f2py's crackfortran has maybe changed so it can't handle the comments? It looks like the f2py call fails because of the exception raised in parsing the fortran.


```
Line #1153 in source/CoordTrans.f:"        call initize ! sets rad, pi used by various routines"
	analyzeline: No name/args pattern found for line.
	Reading file 'source/ntime_max.inc' (format:fix)
	Reading file 'source/variables.inc' (format:fix)
Line #2791 in source/CoordTrans.f:"      REAL*8 xIN(3),xOUT(3),s   ! local vars"
	updatevars: could not crack entity declaration "s!". Ignoring.
	Reading file 'source/AE8_AP8.f' (format:fix,strict)
	Reading file 'source/variables.inc' (format:fix)
	Reading file 'source/ntime_max.inc' (format:fix)
Line #155 in source/AE8_AP8.f:"         CALL INIT_GSM(iyear(isat),idoy(isat),UT(isat),psi) ! compute su"
	analyzeline: No name/args pattern found for line.
	Reading file 'source/variables.inc' (format:fix)
	Reading file 'source/ntime_max.inc' (format:fix)
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.7.1/bin/f2py", line 11, in <module>
    sys.exit(main())
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/f2py/f2py2e.py", line 685, in main
    run_main(sys.argv[1:])
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/f2py/f2py2e.py", line 425, in run_main
    postlist = callcrackfortran(files, options)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/f2py/f2py2e.py", line 329, in callcrackfortran
    postlist = crackfortran.crackfortran(files)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/f2py/crackfortran.py", line 3251, in crackfortran
    readfortrancode(files, crackline)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/f2py/crackfortran.py", line 501, in readfortrancode
    dowithline(finalline)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/f2py/crackfortran.py", line 796, in crackline
    analyzeline(m, pat[1], line)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/f2py/crackfortran.py", line 1184, in analyzeline
    for e in markoutercomma(ll).split('@,@'):
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/f2py/crackfortran.py", line 833, in markoutercomma
    assert not f, repr((f, line, l))
AssertionError: (-1, 'xminp_l = 1.0)   ! minimum lshell to calculate proto', 'xminp_l = 1.0)   ! minimum lshell to calculate proto')
Traceback (most recent call last):
  File "setup.py", line 929, in <module>
    setup(**setup_kwargs)
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/distutils/core.py", line 171, in setup
    return old_setup(**new_attr)
  File "/opt/python/3.7.1/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/numpy/distutils/command/install.py", line 60, in run
    r = old_install.run(self)
  File "/opt/python/3.7.1/lib/python3.7/distutils/command/install.py", line 545, in run
    self.run_command('build')
  File "/opt/python/3.7.1/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 738, in run
    self.compile_irbempy()
  File "setup.py", line 495, in compile_irbempy
    subprocess.check_call(cmd)
  File "/opt/python/3.7.1/lib/python3.7/subprocess.py", line 341, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['f2py', '--overwrite-signature', '-m', 'irbempylib', '-h', 'irbempylib.pyf', 'source/onera_desp_lib.f', 'source/CoordTrans.f', 'source/AE8_AP8.f', 'source/find_foot.f', 'source/drift_bounce_orbit.f', 'only:', 'make_lstar1', 'make_lstar_shell_splitting1', 'find_foot_point1', 'coord_trans1', 'find_magequator1', 'find_mirror_point1', 'get_field1', 'get_ae8_ap8_flux', 'fly_in_nasa_aeap1', 'trace_field_line2_1', 'trace_field_line_towards_earth1', 'trace_drift_bounce_orbit', ':']' returned non-zero exit status 1.
```

**Comment by drsteve** [comment]
Yup. Somewhere in the f2py/crackfortran updates a subtle part of comment handling broke...
That is, on the f77 PARAMETER statement, an inline comment is no longer supported.

To test I moved the inline comments to a single comment line prior to the PARAMETER declarations. Bingo - tests pass.
[https://travis-ci.com/drsteve/spacepy/builds/102700685](https://travis-ci.com/drsteve/spacepy/builds/102700685)

I can initiate a PR to fix this (although I don't know which versions this will fix, my guess would be 1.16.1 and 1.16.2 only, but I'm not sure). Anyone have thoughts on that?
I'm also going to push this change back up to the IRBEM repo on sourceforge.
